docs/pages/server-access/guides/ec2-discovery.mdx

```diff
@@ -0,0 +1,277 @@
+  join_method: iam
+
+  allow:
+  # specify the AWS account which nodes may join from
+  - aws_account: "123456789"
+```
+
+Add the token to the Teleport cluster with:
+```
+tctl create -f token.yaml
+```
+
+More documentation on IAM invite tokens can be found [here](../../management/guides/joining-nodes-aws-iam.mdx)
+
+## Step 2/5. Create an AWS Systems Manager Document
+
+Teleport makes use of AWS Systems Manager documents to execute commands necessary to
+install and start Teleport on discovered EC2 instances.
+
+A document with the following content should be created, named `TeleportDiscoveryInstaller`:
+
+```yaml
+---
+schemaVersion: '2.2'
+description: aws:runShellScript
+parameters:
+  token:
+    type: String
+    description: "(Required) The Teleport invite token to use when joining the cluster."
+  scriptName:
+    type: String
+    description: "(Required) The Teleport instalelr script to use when joining the cluster."
+mainSteps:
+- action: aws:downloadContent
+  name: downloadContent
+  inputs:
+    sourceType: "HTTP"
+    destinationPath: "/tmp/installTeleport.sh"
+    sourceInfo:
+      url: "https://teleport.example.com/webapi/scripts/{{ scriptName }}"
+- action: aws:runShellScript
+  name: runShellScript
+  inputs:
+    timeoutSeconds: '300'
+    runCommand:
+      - /bin/sh /tmp/installTeleport.sh "{{ token }}"
+```
+
+Here, `teleport.example.com` should be substituted with the public proxy URL of the
+Teleport cluster.
+
+When Teleport executes the Systems Manager document it will provide the `token`
+and `scriptName` parameters with defaults.
+
+- `{{ scriptName }}` will by default evaluate to `default-installer`
+- `{{ token }}` will by default evaluate to `aws-discovery-iam-token`
+
+The document `TeleportDiscoveryInstaller` may be named differently
+however the `discovery_service` section must be modified to include
+the `ssm` section. For example:
+
+```yaml
+discovery_service:
+  aws:
+    - types: ["ec2"] 
+      ssm:
+        document_name: "OtherTeleportDiscoveryDocument"
+```
+
+
+## Step 3/5. Set up AWS IAM credentials
+
+In order for Teleport to send commands to EC2 instances, the node
+performing discovery requires an IAM policy with permissions to
+execute Systems Manager commands and retrieve EC2 instances:
+
+```js
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Action": [
+                "ssm:SendCommand",
+                "ec2:DescribeInstances",
+                "ssm:GetCommandInvocation"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+All EC2 instances that are to be added to the Teleport cluster by the
+Discovery Node must include the `AmazonSSMManagedInstanceCore` IAM policy
+in order to receive commands from the Discovery Node. More information
+for IAM best practices on EC2 instances managed by Systems Manager can be found [here](https://aws.amazon.com/blogs/mt/applying-managed-instance-policy-best-practices/)
```

Comment:
We should also move this link to a "Next steps" section. Users won't need to be familiar with IAM best practices to complete this guide, but this could be something they can learn about after completing the guide.